### DISCUSSION TITLE
WT32-SC01 16MB  - Only 4BM reported ....

### DISCUSSION THREAD

**Discussion by DanBendavid** [HASwitchPlate/openHASP#215] [Q&A] [answered]
Hi all,
I have just flashed the WT32-SC01 with the wt32-sc01_full_4MB_v0.6.1.bin.
It is working but only 4MB are reported. How/Where  can i  modify the src to generate a new bin ?
Thank you  

openHASP
--
Version: | 0.6.1
Build DateTime: | Aug 1 2021 23:27:48 UTC
Uptime: | 20s
 

Free Heap: | 93.69 KiB
Free Block: | 93.18 KiB
Fragmentation: | 0
PSRam Free: | 3.99 MiB
PSRam Size: | 3.99 MiB
 

Total: | 48.00 KiB
Free: | 40.35 KiB
Fragmentation: | 2
 

Idle: | off
 

Server: | 192.168.1.10
Username: | mqtt_user
Client ID: | plate_1-c435b0
Status: | Connected
Received: | 0
Published: | 14
Failed: | 7
 

SSID: | IOTMAR1
Signal Strength: | -78dBm (Very bad)
IP Address: | 192.168.1.153
Gateway: | 192.168.1.1
DNS Server: | 192.168.1.1
MAC Address: | C4:DD:57:C4:35:B0
 

Model: | ESP32
Frequency: | 240MHz
Core Version: | v3.3.5-1-g85c43024c
Reset Reason: | CPU0: DEEPSLEEP_RESET / CPU1: EXT_CPU_RESET
Flash Size: | 4.00 MiB
Program Size Used: | 1.50 MiB
Program Size Free: | 1.62 MiB

**Comment by fvanroie** [comment, marked as answer]
You will need to add these 3 lines to the [`wt32-sc01.ini` configuration](https://github.com/HASwitchPlate/openHASP/blob/d97ac75a83e29d293112d4a95ff4d69e3ab34dec/user_setups/esp32/wt32-sc01.ini#L10):

```ini
board_upload.flash_size=16MB
board_upload.maximum_size = 16777216
board_build.partitions = user_setups/esp32/partitions_16MB.csv
```

That should produce a `wt32-sc01_full_16MB_v0.6.1.bin` file after [building the firmware](https://haswitchplate.github.io/openHASP-docs/0.6.1/compiling/local/) with that configuration file.

**Reply by DanBendavid** [reply]
Thank you for your help. It is working. I do not understand why PSRAM is reported at 4MB and not 8MB
<html>
<body>


openHASP
--
Version: | 0.6.2
Build DateTime: | Sep 20 2021 08:15:06 UTC
Uptime: | 58s
 

Free Heap: | 99.41 KiB
Free Block: | 98.33 KiB
Fragmentation: | 1
PSRam Free: | 3.99 MiB
PSRam Size: | 3.99 MiB
 

Total: | 48.00 KiB
Free: | 40.06 KiB
Fragmentation: | 2
 

Idle: | off
 

Server: | 192.168.1.10
Username: | mqtt_user
Client ID: | plate_1-c435b0
Status: | Connected
Received: | 11
Published: | 14
Failed: | 0
 

SSID: | .........
Signal Strength: | -75dBm (Weak)
IP Address: | 192.168.1.153
Gateway: | 192.168.1.1
DNS Server: | 192.168.1.1
MAC Address: | C4:DD:57:C4:35:B0
 

Model: | ESP32
Frequency: | 240MHz
Core Version: | v3.3.5-1-g85c43024c
Reset Reason: | CPU0: DEEPSLEEP_RESET / CPU1: EXT_CPU_RESET
Flash Size: | 16.00 MiB
Program Size Used: | 1.59 MiB
Program Size Free: | 2.00 MiB


</body>
</html>

**Reply by fvanroie** [reply]
> I do not understand why PSRAM is reported at 4MB and not 8MB

It's a limitation of the ESP32 that up to 4 MB of PSRAM can be mapped into CPU data memory space at a time. As only 4 MB memory space is considered for External SRAM, the other 4MB needs to be [mapped in-and-out](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/himem.html) manually by the application. openHASP doesn't support that upper 4MB usage.
